### PR TITLE
New documentation on App Service debugging, and some docs organizational changes

### PR DESCRIPTION
**Pull request by pamelafox**
## Purpose

This pull request includes changes to the `README.md` and `docs/` files to improve the documentation of the `azure-search-openai-demo` project. The most important changes include:

* adding a brand new doc on debugging App Service deployments
* reorganizing the sections in the `README.md`
* moving the images to a new `docs/images/` directory
* turning the FAQs into docs
* adding an index page for the additional docs

I'll add inline comments on relevant changes to explain motivation. Feedback welcome!

## Does this introduce a breaking change?

```
[X] Yes - @mattgotteiner will need to update any links that point to the login doc since I moved it into docs for consistency, like in the blog post.
[ ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[X] Documentation content changes
[ ] Other... Please describe:
```

## How to Test

* Click new or changed links

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,10 +144,11 @@ Then bring down the project code:
 1. Run `azd init -t azure-search-openai-demo`
     * note that this command will initialize a git repository and you do not need to clone this repository
 
-### Deploying from scratch
+### Deploying
 
-Execute the following command, if you don't have any pre-existing Azure services and want to start from a fresh deployment.
+Follow these steps to provision Azure resources and deploy the application code:
 
+1. (Optional) This is the point where you can customize the deployment by setting environment variables, in order to [use existing resources](#using-existing-azure-resources), [enable optional features](#enabling-optional-features), or [deploy to free tiers](docs/deploy_lowcost.md).
```
Comment (pamelafox) [inline README.md:151]: My thought here is to call out that any customization should happen first. My other thought is that we could move "using existing" and "enabling optional features" into its own doc, to significantly shorten this README if we wanted. But that's a larger change, so I just made this one.

File: README.md [hunk 2]
```diff
@@ -242,6 +238,13 @@ either you or they can follow these steps:
 
 ## Enabling optional features
 
+### Using GPT-4
```
Comment (pamelafox) [inline README.md:241]: This was an FAQ, moved it up here. FAQs are hard to link to, the way we implemented them, sections with proper headings are more usable.

File: README.md [hunk 3]
```diff
@@ -317,15 +320,6 @@ This sample is designed to be a starting point for your own production applicati
 but you should do a thorough review of the security and performance before deploying
 to production. Read through our [productionizing guide](docs/productionizing.md) for more details.
 
-## Resources
```
Comment (pamelafox) [inline README.md:320]: Moved lower

File: README.md [hunk 4]
```diff
@@ -336,94 +330,7 @@ To clean up all the resources created by this sample:
 
 The resource group and all the resources will be deleted.
 
-
-### FAQ
-
-<details><a id="compare-samples"></a>
```
Comment (pamelafox) [inline README.md:342]: All FAQs became README sections or standalone docs

File: README.md [hunk 5]
```diff
@@ -437,7 +344,19 @@ Here are the most common failure scenarios and solutions:
...
+## Resources
+
+* [Additional documentation for this app](docs/README.md)
+* [📖 Revolutionize your Enterprise Data with ChatGPT: Next-gen Apps w/ Azure OpenAI and AI Search](https://aka.ms/entgptsearchblog)
+* [📖 Azure AI Search](https://learn.microsoft.com/azure/search/search-what-is-azure-search)
+* [📖 Azure OpenAI Service](https://learn.microsoft.com/azure/cognitive-services/openai/overview)
+* [📖 Comparing Azure OpenAI and OpenAI](https://learn.microsoft.com/azure/cognitive-services/openai/overview#comparing-azure-openai-and-openai/)
+* [📖 Access Control in Generative AI applications with Azure Cognitive Search](https://techcommunity.microsoft.com/t5/ai-azure-ai-services-blog/access-control-in-generative-ai-applications-with-azure/ba-p/3956408)
+* [📺 Quickly build and deploy OpenAI apps on Azure, infused with your own data](https://www.youtube.com/watch?v=j8i-OM5kwiY)
+* [📺 AI Chat App Hack series](https://www.youtube.com/playlist?list=PL5lwDBUC0ag6_dGZst5m3G72ewfwXLcXV)
```
Comment (pamelafox) [inline README.md:358]: Added playlist from hackathon

File: README.md [hunk 6]
```diff
@@ -437,7 +344,19 @@ Here are the most common failure scenarios and solutions:
...
 
-1. After running `azd up` and visiting the website, you see a '404 Not Found' in the browser. Wait 10 minutes and try again, as it might be still starting up. Then try running `azd deploy` and wait again. If you still encounter errors with the deployed app, consult these [tips for debugging App Service app deployments](http://blog.pamelafox.org/2023/06/tips-for-debugging-flask-deployments-to.html) or watch [this video about downloading App Service logs](https://www.youtube.com/watch?v=f0-aYuvws54). Please file an issue if the logs don't help you resolve the error.
+1. After running `azd up` and visiting the website, you see a '404 Not Found' in the browser. Wait 10 minutes and try again, as it might be still starting up. Then try running `azd deploy` and wait again. If you still encounter errors with the deployed app, consult the [guide on debugging App Service deployments](docs/appservice.md). Please file an issue if the logs don't help you resolve the error.
+
+## Resources
+
+* [Additional documentation for this app](docs/README.md)
```
Comment (pamelafox) [inline README.md:351]: Added link to docs index

File: docs/other_samples.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
+# Alternative RAG chat samples
+
```
Comment (pamelafox) [inline docs/other_samples.md:2]: Moved from FAQ

**Comment by pamelafox** [comment]
cc @charris-msft Some of this was inspired by user studies. Particularly my comment about potential future size reductions of the README.

**Comment by dfberry** [comment]
> @dfberry Curious if you have any feedback as well. And wanted to make sure this didnt break any links from your article (not sure if you link to any subheadings).

I'm not dependent on the docs in the repo for _Learn_ content - I appreciate you letting me know though. The root Readme, from what I've seen in UX and metrics, should be pretty slim covering just how to use the thing. You can see that in some of the Azure SDK work for Identity.

**Comment by pamelafox** [comment]
@dfberry Thanks for confirming! That's what I suspected, our README is not what one could call pretty slim. Am curious what @charris-msft and @mattgotteiner think about my suggestion of moving the 'azd env set' stuff to separate docs.

**Comment by charris-msft** [comment]
> @dfberry Thanks for confirming! That's what I suspected, our README is not what one could call pretty slim. Am curious what @charris-msft and @mattgotteiner think about my suggestion of moving the 'azd env set' stuff to separate docs.

After hearing some other feedback this week about the pudgy README, I'm 100% in favor of identifying portions of the content that can logically be moved into their own docs.

**Review by charris-msft — APPROVED** [review]
LGTM!

**Comment by pamelafox** [comment]
Merged recent changes from @mattgotteiner to login doc from main.

File: README.md (hunk 1 above)
Comment (mattgotteiner) [reply README.md:152]: I'm sold, let's do it

**Review by mattgotteiner — APPROVED** [review]

**Pull request merged**